pkg/storaged/client.js

```diff
@@ -1772,6 +1772,8 @@ client.export_mount_point_mapping = () => {
                 const dir = client.strip_mount_point_prefix(utils.decode_filename(c[1].dir.v));
                 if (dir)
                     mpm[dir] = utils.decode_filename(b.PreferredDevice);
+                if (b.IdType === "swap")
+                    mpm[b.IdType] = utils.decode_filename(b.PreferredDevice);
```

Comment:
This means that there can be only one swap, right? Let's change the whole data structure to not be restrictive like that. Maybe have a mapping from device to mount point, swap, whatever?

On the other hand, why doesn't Anaconda detect swaps itself?  We need this `cockpit_mount_points` thing because Anaconda has no other way to learn about the mount points (and it doesn't want to read /etc/fstab). But Anaconda can look at udev properties, no?

Reply:
There might be another swap partition but for another OS installation on the same disk I guess. How can anaconda then know whose is the swap? I guess only through /etc/fstab.

Reply:
Ah, that makes sense. But then you need to check `c[1].type`, not IdType.

Reply:
Clear let me redo.